src/codegen.cpp

```diff
@@ -7017,11 +7018,12 @@ static Function* gen_cfun_wrapper(
     ctx.builder.SetCurrentDebugLocation(noDbg);
     allocate_gc_frame(ctx, b0, true);
 
-    Value *world_age_field = get_last_age_field(ctx);
+    ctx.world_age_field = get_last_age_field(ctx);
     jl_aliasinfo_t ai = jl_aliasinfo_t::fromTBAA(ctx, ctx.tbaa().tbaa_gcframe);
     Value *last_age = ai.decorateInst(
-            ctx.builder.CreateAlignedLoad(ctx.types().T_size, world_age_field, ctx.types().alignof_ptr));
+            ctx.builder.CreateAlignedLoad(ctx.types().T_size, ctx.world_age_field, ctx.types().alignof_ptr));
 
+    cast<LoadInst>(last_age)->setMetadata(LLVMContext::MD_invariant_group, llvm::MDNode::get(ctx.builder.getContext(), {}));
```

Comment:
I think this metadata is incorrect - this is the world age loaded _before_ the OC-related stores below, so it's not the same value as `Base.tls_world_age()` within the function

Reply:
I guess we want to add strips to any of the stores to the world age pointer?

Reply:
Though I'm not sure what the annotation says about visible stores into the pointer

Reply:
IIUC the annotation only says anything about annotated stores/loads, so it's enough to leave this load and those stores unannotated